python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -1335,6 +1335,29 @@ def _get_proxy_base_class(cls):
     return object
 
 
+def as_proxy_object(obj: Any) -> Any:
+    """
+    Wraps a cudf or pandas object in a proxy object if applicable.
```

Comment:
The docs could clarify a few things. I do not know the memory semantics of this code path.

Will the proxied object be physically stored on GPU or CPU? Does that depend on the input type (pd/cudf)?

Does this create a copy or just "wrap" the original object?

Reply:
It doesn't create a copy, it just wraps around the original object keeping it on it's original memory resource either GPU / CPU.

Reply:
Do we need to document (or test) that modifying the new proxy object also modifies the original object?

Reply:
Done